sql/src/sparksql_upgrade/rules.py

```diff
@@ -460,7 +461,12 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
 
             if expression_segment is not None:
                 expression_child = expression_segment.children().first()
-                if expression_child[0].type == "function":
+                # cast can either be a keyword or a function depending on if were iterating on
+                # parsed on updated code.
+                if expression_child[0].type == "keyword":
```

Comment:
Nice catch! The original lint result contains a keyword rather than a function expression so this makes sense to me. Have you verified that when calling `sqlfluff.fix` that we do not infinitely recurse with this change?

Reply:
Yup, or well more accurately without the fix sqlfluff does recurse until it hits the recursion limit (10 by default I believe).